src/cargo/ops/cargo_generate_lockfile.rs

```diff
@@ -106,4 +121,23 @@ pub fn update_lockfile(manifest_path: &Path,
             None => {}
         }
     }
+
+    fn compare_dependency_graphs<'a>(previous_resolve: &'a Resolve,
+                                     resolve: &'a Resolve) ->
+                                     Vec<(Option<&'a PackageId>, Option<&'a PackageId>)> {
+        let mut changes = HashMap::new();
+        for dep in previous_resolve.iter() {
+            changes.insert(dep.name(), (Some(dep), None));
+        }
+        for dep in resolve.iter() {
+            if !changes.contains_key(dep.name()) {
+                changes.insert(dep.name(), (None, None));
+            }
+            let value = changes.get_mut(dep.name()).unwrap();
+            value.1 = Some(dep);
+        }
+        let mut package_names: Vec<&str> = changes.keys().map(|x| *x).collect();
+        package_names.sort();
+        package_names.iter().map(|name| *changes.get(name).unwrap()).collect()
```

Comment:
I believe you can just use `changes[name]` here instead of `changes.get(name).unwrap()`